source/MaterialXRuntime/Commands/ConnectionCommands.h

```diff
@@ -0,0 +1,65 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_RTCONNECTIONCOMMANDS_H
+#define MATERIALX_RTCONNECTIONCOMMANDS_H
+
+/// @file
+/// Commands for connection handling.
+
+#include <MaterialXRuntime/RtCommandEngine.h>
+#include <MaterialXRuntime/RtAttribute.h>
+
+namespace MaterialX
+{
+
+class RtMakeConnectionCmd : public RtCommandBase
+{
+public:
+    RtMakeConnectionCmd(const RtOutput& src, const RtInput& dest) :
+        _src(src),
+        _dest(dest)
+    {}
+
+    static RtCommandPtr create(const RtOutput& src, const RtInput& dest);
+
+    void execute(RtCommandResult& result) override;
+    void undo(RtCommandResult& result) override;
+
+private:
+    RtOutput _src;
+    RtInput _dest;
+};
+
+class RtBreakConnectionCmd : public RtCommandBase
+{
+public:
+    RtBreakConnectionCmd(const RtOutput& src, const RtInput& dest) :
+        _src(src),
+        _dest(dest)
+    {}
+
+    static RtCommandPtr create(const RtOutput& src, const RtInput& dest);
+
+    void execute(RtCommandResult& result) override;
+    void undo(RtCommandResult& result) override;
+
+private:
+    RtOutput _src;
+    RtInput _dest;
+};
+
+namespace RtCommand
```

Comment:
Common namespace for all runtime command wrapper functions. These functions will create command objects internally and pass to the command engine for execution and undo queuing, so users don't need to care about that.

Reply:
This is nice and clean. I guess in this model the engine is the only one which owns a reference to a command.

What happens in the case if someone creates a command and just calls execute() ?  Seems the exact same thing occurs, but the caller can own a reference as well and these commands are not part of the undo queue, in which case ordering is not maintained?

Would it be better that the individual commands interfaces not be public ?

Reply:
That's a good point. I will try to move the command classes to the private API, I think that would work out well.